### ISSUE TITLE
Rule proposal: Use explicit-function-return-type but skip void

### ISSUE THREAD

**Issue by klutt** [typescript-eslint/typescript-eslint#4064]
### Description

I was looking at the rule explicit function return type, but there is an option I am missing. I'd like to be allowed to skip specifying `void` for void functions or methods. It should work by checking all return statements in the function/method and warn if:

- A return statement is found, AND
- That return statement returns an object, AND
- The function or method either does not have a specified return type, or is specified to `void` or `Promise<void>`

The reason is that one might think that it's overly verbose to specify the return type for void functions. But if you encounter such a function, how do you know if it's void or unspecified without analyzing it?

### Fail

```ts
function foo() {
    return 42; // FAIL: Returning an object without specification
}

function bar(): void {
    return 42; // FAIL: Returning an object for void function
}
```

### Pass

```ts
function foo() {
    return; // Ok: Since returntype is unspecified, void is assumed
}

function bar(): void {
    return; // Ok
}
```
